synchronizer/actions/etrog/processor_l1_sequence_batches.go

```diff
@@ -450,6 +384,7 @@ func (p *ProcessorL1SequenceBatchesEtrog) checkTrustedState(ctx context.Context,
 		if p.sync.IsTrustedSequencer() {
 			log.Errorf("TRUSTED REORG DETECTED! Batch: %d reson:%s", batch.BatchNumber, reason)
 			p.halt(ctx, fmt.Errorf("TRUSTED REORG DETECTED! Batch: %d", batch.BatchNumber))
+			panic("TRUSTED REORG DETECTED: halt function never have to finish! it blocks the process")
```

Comment:
this panic is unused code. This code won't be executed. Why to put this line? Please remove it

Reply:
This panic have to reason to be there: 
- As a safeguard: if, for some bug, `halt` funcion return is going to prevent to execute more code avoiding a potential risk 
- For unittest: there are no way to test a infite loop function, so the way of doing that is mocking halt object and requiring that produce a panic